mu.rb

```diff
@@ -26,13 +26,25 @@ def log
     @log
   end
 
-  def query(query)
+  def query(query, **options)
     log.info "Executing query: #{query}"
-    sparql_client.query query
+    @sparql_client = sparql_client(**options)
+    @sparql_client.query query
   end
 
-  def sparql_client
-    options = {}
+  def sparql_client(**options)
+    if Mu::truthy? options[:sudo]
+      if Mu::truthy? ENV['ALLOW_MU_AUTH_SUDO']
+        options[:headers] = { 'mu-auth-sudo': 'true' }
+      else
+        log.error "Error, sudo request but service lacks ALLOW_MU_AUTH_SUDO header"
```

Comment:
Shouldn't we also raise an error here instead of only logging one?  Query will now still be executed with a non-sudo SPARQL client.

Reply:
Should I go with `raise`?

```ruby
raise "Error, sudo request but service lacks ALLOW_MU_AUTH_SUDO header"
```

Reply:
Yes, seems more in line with the JS template to me :+1: